app/javascript/components/applications/application_form/PaperApplicationForm.js

```diff
@@ -79,7 +80,8 @@ class PaperApplicationForm extends React.Component {
       alert('There was an error on submit. Please check values and try again.')
     }
 
-    this.setState({ loading: false })
+    this.setState({ submittedValues, loading: true, failed: false })
```

Comment:
looks like loading is set to true at the top of this function, do we need to set it again? are we setting it to be false anywhere?